### ISSUE TITLE
Images Fail to Download to External Storage

### ISSUE THREAD

**Issue by gozes** [jmshrv/finamp#955]
![finamp-logs.txt](https://github.com/user-attachments/files/17758508/finamp-logs.txt)

So I joined the beta today and trying to download some of my library to the SD card on my DAP. However as you can see from the logs it looks like I'm getting a lot of failed image downloads. Looking at the SD card it looks like the songs downloaded fine but I haven't tried all of them to be sure. It's hard for me to tell if the missing images would cause any issues. I usually add the cover art to every song when I tag them so it's unclear to me what images it's trying to download

**Comment by gozes** [comment]
So tapping around it seems that the images that fail to download are the ones that Wil be used in the player and as the album cover as songs Sam's to play fine. Reducing the number of runners from 10 to 15 didn't really do much aside from making the process slower

**Comment by Chaphasilor** [comment]
Yes, that is a known issue that I also mentioned in my comment https://github.com/jmshrv/finamp/issues/842#issuecomment-2473997853

I've managed to pinpoint this to images failing to download for any tracks where no other track from that album has already been downloaded. It seems to be a permission issue, but I'm not yet totally sure why it happens.

Finamp doesn't look at the tags of files at all, all information is fetched from the server. That's by design, and in some cases like when transcoding is the only way to get the data.

**Comment by gozes** [comment]
@Chaphasilor so it seams that even after given it the permissions by adb it still fails to get images. suggestions?

**Comment by Chaphasilor** [comment]
Maybe @Komodo5197 could take a look at this, they have more experience with the download system. I'd imagine it's a rather simple issue, since downloading the tracks itself works fine.

**Comment by gozes** [comment]
Let know what support you would need from my side. Happy to help get this
fix :)
.

**Comment by Komodo5197** [comment]
It seems like android permissions refuse to allow any file without a valid music file extension to be written to the /Music folder.  If you use the /Documents directory, or write all the images as .mp3, everything works fine.

**Comment by gozes** [comment]
> It seems like android permissions refuse to allow any file without a valid music file extension to be written to the /Music folder. If you use the /Documents directory, or write all the images as .mp3, everything works fine.

All my music is in flac. All the music is going into `/Music/finamp` in the sd card

**Comment by gozes** [comment]
Yup, deleted everything and set the location to `/Downloads` and the images started to show up

**Comment by Chaphasilor** [comment]
I'm wondering if it's best to just always save the images to the internal location, and only save the actual tracks to the custom ones.  
Not sure how complicated that would be, but given that it already works when downloading on track to the internal location and the remaining tracks to a custom one, it should be doable.

Saving the images with a different extension kind of defeats the purpose of saving them in a user-accessible location anyway.

**Comment by gozes** [comment]
Hmm that may get troublesome with a large library. For example my hiby m300
has tiny tiny amounts of internal storage but It can take up to a 2TB micro
SD card. That said if storing images for 2TB worth of music only comes up
to maybe a few hundred megabytes then maybe it makes sense
.

**Comment by Chaphasilor** [comment]
We've put up a disclaimer now for custom locations to avoid the `Music` directory, that should hopefully resolve the issue. There's isn't a whole lot more that we can do to properly fix this without weird workarounds...

**Comment by Seedling1154** [comment]
Easy work around if anyone stumbles on this. Go to download settings and select your internal storage as favorite than go back and click on Cache current library images this will download only the covers to your internal storage than switch back to your SD Card and download your music like normal should have the covers then.
